src/Exporter/ContentExporter.php

```diff
@@ -31,7 +31,8 @@ public function exportEntity(ContentEntityInterface $entity, array $context = []
     $context += [
       'content_sync' => TRUE,
     ];
-
+    // Allows normalizers to know that this is a content sync generated entity.
+    $entity->is_content_sync = TRUE;
```

Comment:
Dynamic properties are deprecated in PHP 8.2. Given the flag in `$context`, can probably just make use of it?...